runtime/compiler/control/CompilationController.cpp

```diff
@@ -203,7 +203,14 @@ TR_OptimizationPlan *TR::DefaultCompilationStrategy::processEvent(TR_MethodEvent
          // For sync re-compilation we have attached a plan to the persistentBodyInfo
          bodyInfo = TR::Recompilation::getJittedBodyInfoFromPC(event->_oldStartPC);
          methodInfo = bodyInfo->getMethodInfo();
-         if (bodyInfo->getUsesJProfiling() && !bodyInfo->getIsProfilingBody())
+
+         if (methodInfo->getReasonForRecompilation() == TR_PersistentMethodInfo::RecompDueToInlinedMethodRedefinition)
```

Comment:
Could we combine the code here? It is the same logic for RecompDueToInlinedMethodRedefinition and TR_PersistentMethodInfo::RecompDueToJProfiling. Previously, we had to guess based on the type of the body because we didn't have the reason field.
Similarly, now that you passed the reason for GCR, we could use that test instead of testing `bodyInfo->getUsesGCR()`

Reply:
Thanks for the suggestion. modified the change.